class-udm-updater.php

```diff
@@ -423,6 +424,22 @@ public function puc_retain_fields($f) {
 		return $f;
 	}
 
+	/**
+	 * WordPress filter puc_request_info_result-(slug)
+	 * In this filter, we disconnect the site that has no license after receiving the plugin update information from the updates server.
+	 *
+	 * @param YahnisElsts\PluginUpdateChecker\v5p5\Plugin\PluginInfo $info - the plugin update info from the updates server
+	 *
+	 * @return YahnisElsts\PluginUpdateChecker\v5p5\Plugin\PluginInfo
+	 */
+	public function puc_request_info_result($info) {
+		if ($this->is_connected() && empty($info->toStdClass()->extraProperties['x-spm-connected'])) {
+			$this->disconnect();
+			if ($this->debug) error_log("udm_updater: ".$this->slug." - the site was disconnected because its license entitlement has been removed.");
```

Comment:
@hisman This message is too specific and may be incorrect. Perhaps the entitlement isn't removed, but is being used on another site.